change the logic of show more from number of lines to number characters

KaleemNeslit opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: #9276 
## Description of Changes
- Change the "show more " from number of lines to  length Or character .
- currently the length of content is 500 character  output is attached below for better understanding 


![image](https://github.com/user-attachments/assets/f947fe4c-3e61-4d9c-ae8d-d3b3ee7dced2)

timolegros commented on packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.tsx:
```diff
@@ -209,7 +209,7 @@ export const ThreadCard = ({
               <QuillRenderer
                 doc={bodyText}
-                cutoffLines={4}
+                maxChars={500}
```
Can we maybe put this as a constant in `shared/src/constants.ts`? It is a useful reference for me on the backend where we will soon be truncating all bodies longer than the amount required for previews.

KaleemNeslit replied:
sure , i will put there . 
"I noticed that we're using <QuillRenderer /> in different places, and in some instances, we're setting cutoffLines to 50 or 10. Should I replace all occurrences of cutoffLines with maxchars, or should we keep support for cutoffLines so that it continues to work as it currently does in other places?"
@Israellund

Israellund replied:
@KaleemNeslit I think this is a good question for @mzparacha What do you think?

KaleemNeslit replied:
should i add 5000 character for 50 cutoffLines and 1000 character for 10 lines @Israellund ?

Israellund replied:
@KaleemNeslit See above comment by Sachin

KaleemNeslit commented:
https://github.com/user-attachments/assets/6588669d-a745-47e9-9ccf-7c8e7bfb0dc3

after resolving the conflict

masvelio commented:
@KaleemNeslit what's the status here? Please make sure that PRs do not getting stale. It's your responsibility to merge it as soon as possible 🙏

masvelio commented:
@KaleemNeslit

KaleemNeslit commented:
> @KaleemNeslit

@masvelio , this PR needs review. I have addressed the comments and requested reviews from 4 people, but 2 are still pending, even though it's been many days.

KaleemNeslit commented:
@masvelio  @Israellund  @burtonator  @mzparacha  Whenever you have a chance, could you please review my PR? It's been in a stale state for a few days. Thanks!

masvelio commented:
@mzparacha could you take a look on this one?

mzparacha commented:
Noticed some inconsistencies b/w the master and the current implementation.

Current:
https://github.com/user-attachments/assets/992ccb5d-4f05-488e-b5be-7b53ff659c11

Master:
https://github.com/user-attachments/assets/780b1e78-1a82-453b-8e34-e59e96e781da

Just for more context: this quill text cutoff has historically caused many inconsistencies b.w devices (iOS safari for instance https://github.com/hicommonwealth/commonwealth/issues/8611) and thread bodies (some of which you can see in the `master` branch video above). Lets aim to fix the multiline `\n` and markdown issues (plz compare the threads in videos for reference).


cc: @KaleemNeslit

KaleemNeslit commented:
> Noticed some inconsistencies b/w the master and the current implementation.
> 
> Current:
> 
>  Screen.Recording.2024-10-25.at.6.37.35.PM.mov 
> Master:
> 
>  Screen.Recording.2024-10-25.at.6.39.04.PM.mov 
> Just for more context: this quill text cutoff has historically caused many inconsistencies b.w devices (iOS safari for instance #8611) and thread bodies (some of which you can see in the `master` branch video above). Lets aim to fix the multiline `\n` and markdown issues (plz compare the threads in videos for reference).
> 
> cc: @KaleemNeslit
yes in master we are showing more base on number of line cutoff , if 6 so after 6 \n it will show "show more "
so we change the logic inti character length . if you have 100 character it will show "show more " it does not matter how many /n or line you have . 
@mzparacha  what should we do in that case ?

mzparacha commented:
Lets aim to fix the multiline \n and markdown issues (plz compare the threads in videos for reference).

KaleemNeslit commented:
updates we need to update some logic related to the cutofflines and maxChars so i am converting PR  as draft to implement the requirements

timolegros commented on libs/shared/src/constants.ts:
```diff
@@ -39,3 +39,7 @@ export const MAX_TRUNCATED_CONTENT_LENGTH = 2_000;
+
+export const MIN_CHARS_TO_SHOW_MORE = 500;
+export const MAIDEN_CHARS_TO_SHOW_MORE = 1000;
+export const MAX_CHARS_TO_SHOW_MORE = 5000;
```
The maximum length of bodies stored in the database is 2000. See `MAX_TRUNCATED_CONTENT_LENGTH` above. Any threads that contain more than 2000 characters have the full body stored on R2 and this body is not loaded until the user opens the thread page. May need to reduce this value to 2000 and check if `content_url` is set to know if there are more than 2k characters.

I think it is important for @mzparacha to review this PR since he worked on the client side for loading thread bodies that are longer than 2k characters.
